### PR TITLE
integration: add regression test for LP: #1541317

### PR DESCRIPTION
**Pull request by mvo5**
This test ensures that selecting a channel is stored correctly. This is a regression test for https://bugs.launchpad.net/software-center-agent/+bug/1541317

### CODE REVIEW

**Comment by mvo5** [comment]
retest this please

**Comment by niemeyer** [comment]
Looks good to me. Adding an item to my sprint list for us to talk about the syntax of "hello-world/edge".

**Comment by mvo5** [comment]
Please do not merge just yet just yet because the store is currently still broken. Once the store is fixed we can merge this to ensure we get early warning in case this issues comes back.

**Comment by chipaca** [comment]
:+1:

**Comment by mvo5** [comment]
retest this please

**Comment by elopio** [comment]
retest this please

File: integration-tests/tests/installApp_test.go [hunk 1]
```diff
@@ -89,3 +89,15 @@ func (s *installAppSuite) TestInstallUnexistingAppMustPrintError(c *check.C) {
 			"unexisting failed to install: snappy package not found\n",
 		check.Commentf("Wrong error message"))
 }
+
+func (s *installAppSuite) TestInstallFromStoreMetadata(c *check.C) {
+	cmd := exec.Command("sudo", "snappy", "install", "hello-world/edge")
```
Comment (elopio) [inline integration-tests/tests/installApp_test.go]: Any reason for not using cli.ExecCommand?

File: integration-tests/tests/installApp_test.go [hunk 2]
```diff
@@ -89,3 +89,15 @@ func (s *installAppSuite) TestInstallUnexistingAppMustPrintError(c *check.C) {
...
+	_, err := cmd.CombinedOutput()
+	c.Check(err, check.IsNil)
+	defer common.RemoveSnap(c, "hello-world")
+
+	cmd = exec.Command("sudo", "snappy", "info", "hello-world")
```
Comment (elopio) [inline integration-tests/tests/installApp_test.go]: same question.

Comment (elopio) [inline integration-tests/tests/installApp_test.go]: And snappy info shouldn't require sudo.

File: integration-tests/tests/installApp_test.go (hunk 1 above)
Comment (mvo5) [inline integration-tests/tests/installApp_test.go]: Because I'm a mupett and copied the commands from the previous test. cargo-culting ftw!

**Comment by mvo5** [comment]
Thanks @elopio your suggestion to use the cli and common helpers make a lot of sense. I added that now.

File: integration-tests/tests/installApp_test.go (hunk 1 above)
Comment (elopio) [inline integration-tests/tests/installApp_test.go]: Ah, the one above was written before we had support for expected errors. I'll update that one.

**Comment by elopio** [comment]
retest this please

**Comment by elopio** [comment]
:+1: Waiting for travis to merge.

**Pull request merged**